includes/Checker/Checks/Abstract_PHP_CodeSniffer_Check.php

```diff
@@ -74,6 +74,18 @@ final public function run( Check_Result $result ) {
 		// Backup the original command line arguments.
 		$orig_cmd_args = $_SERVER['argv'];
 
+		/*
+		 * Reset \PHP_CodeSniffer\Config::$overriddenDefaults to prevent
+		 * incorrect results when running multiple checks.
+		 *
+		 * PHPStan ignore reason: PHPStan raised an issue because we can't
+		 * use class in ReflectionClass.
+		 *
+		 * @phpstan-ignore-next-line
+		 */
+		$reflected_phpcs_config = new \ReflectionClass( '\PHP_CodeSniffer\Config' );
+		$reflected_phpcs_config->setStaticPropertyValue( 'overriddenDefaults', array() );
+
```

Comment:
@mukeshpanchal27 I see this is working for some tests but not others.

To resolve that first issue you could look at wrapping this in a `class_exists` conditional.

Secondly, as this is only needed to resolve the issues in tests we shouldn't include this in the production code here. Can we explore create a `Static_Check_UnitTestCase` similar to the [`Runtime_Check_UnitTestCase`](https://github.com/10up/plugin-check/blob/trunk/tests/phpunit/utils/TestCase/Runtime_Check_UnitTestCase.php) and including this in a setup method? I'm not 100% sure this would work but I'm against including this code here just to fix unit tests.

cc @joemcgill

Reply:
@mukeshpanchal27 as well as a class exists check we will also want to include a check for the property as well.

Reply:
@Joe Grainger, I have made slight updates to the code to ensure compatibility with PHP 7.X, and now all the tests are passing. ✅